src/encoded/upgrade/analyte.py

```diff
@@ -0,0 +1,13 @@
+from typing import Any, Dict
+
+from snovault import upgrade_step
+
+
+@upgrade_step("analyte", "1", "2")
+def upgrade_analyte_1_2(value: Dict[str, Any], system: Dict[str, Any]) -> None:
+    """Change `components` to `molecule_detail`."""
+    components = value.get("components")
+    if components is not None:
+        if components:
+            value["molecule_detail"] = components
+        del value["components"]
```

Comment:
I get you're trying to be super safe here but just FYI
```
components = value.get("components")
if components:
    value["molecule_detail"] = components
    del value["components"]
```
is functionally equivalent since in the previous type definition you are ensuring there is at least 1 element in the list within the enum (since the empty list is the only non-null False value that would get you in a situation where you don't want to propagate the value but still need to del it).